### ISSUE TITLE
How to use sql subcommand exactly?

### ISSUE THREAD

**Issue by happydentist** [aotimme/gocsv#18]
I use ON WINDOW10
gocsv sql --query "SELECT column FROM sample" sample.csv
but in vain.
The CMD window show
Error:Binary was complied with 'CGO_ENABLED=0' go-sqlite3 requires cgo to work. this is a stub

**Comment by flowrean** [comment]
I have the same behaviour on Linux. But on macOS it does work for me 🤔

**Comment by flowrean** [comment]
In my own fork, I have been building from macOS and noticed that the sql subcommand also does not work with the Linux binary. Following the documentation of go-sqlite3, I tried to cross compile using xgo (which worked after tackling #30) and now SQL works again in the resulting Linux binary. Which you can verify in [my latest release](https://github.com/flowrean/goscsv/releases/tag/latest).
@aotimme Does this ring a bell with you? Would it be desirable and possible to always use xgo to build releases?

**Comment by gimmesomethinggood** [comment]
@flowrean Do the binaries for linux have the fix to run sql subcommand?  Or do I need to compile from source?  The reason why I ask is that the latest is saying "released this on Nov 15, 2019".  Sorry for my ignorance if this is already in the binary....

**Comment by flowrean** [comment]
@gimmesomethinggood To release, I follow the procedure from [DEV.md](https://github.com/aotimme/gocsv/blob/04a6b585bb9cfb976665ca7a1417508e9971ca69/DEV.md) and that includes overwriting the same 'latest' release each time. So the binaries have indeed been updated there (being more recent than the 'release' itself). But beware, my version uses semicolon as default delimiter. That should be the only difference.
You can build from source, but (in my experience) whether it will be fixed depends not on a change to the source code, but on the way you build it. Try these commands:
```shell
export CGO_ENABLED=1
go get -u github.com/karalabe/xgo
go get -u github.com/mattn/go-sqlite3
go get -u github.com/aotimme/gocsv
xgo -ldflags "-X github.com/aotimme/gocsv/cmd.VERSION=x.x.x -X github.com/aotimme/gocsv/cmd.GIT_HASH=92440509cb0c9c7dc186ada3f62c4ba1e97fbe41" github.com/aotimme/gocsv/
```
(only tested build on macOS and resulting Linux binary)

**Comment by aotimme** [comment]
Hey @flowrean. Re
> @aotimme Does this ring a bell with you? Would it be desirable and possible to always use xgo to build releases?

It doesn't ring a bell. I've been using the cross-compiled `gocsv` on ubuntu boxes but I guess I just hadn't used the `sql` subcommand there so I hadn't noticed this issue.

I think `xgo` sounds like a great idea. Can you check out the following pull request to see if this looks good to you? #32 

If it looks good I can merge it and compile and release. Once it's out there (or ideally before!) it would be helpful if someone with access to a Windows box (or willing to use VirtualBox, etc.) could verify that the cross-compiled Windows binaries work with the `sql` subcommand.

**Comment by aotimme** [comment]
This should be fixed now thanks to @flowrean's work and #32 and #33. The Windows releases at https://github.com/aotimme/gocsv/releases should now work for the `sql` subcommand.

I'm going to close this issue. Please create a new issue if you run into issues.
